client/src/scenes/GameScene.ts

```diff
@@ -0,0 +1,224 @@
+import Phaser, { Math as pMath, Scene } from "phaser";
+import { ClientMessageType } from "../../../common/messages";
+import { Bullet, Direction, GameState, Player } from "../../../common/types";
+import { InterpolationBuffer } from "interpolation-buffer";
+import { RoomConnection } from "../connection";
+import { MAP, BG_COLOR } from '../../../common/map';
+
+class GameScene extends Scene {
+  private connection!: RoomConnection;
+
+  // The buffer which holds state snapshots
+  private stateBuffer: InterpolationBuffer<GameState> | undefined;
+  // A map of player sprites currently connected
+  private players: Map<string, Phaser.GameObjects.Sprite> = new Map();
+  // A map of bullet sprites currently in-air
+  private bullets: Map<number, Phaser.GameObjects.Sprite> = new Map();
+  // The graphics representation of the map objects
+  private mapGfx: Phaser.GameObjects.Graphics | undefined;
+  // The Hathora user for the current client's connected player
+  private hathoraUser: object & { id: string } | undefined;
+  // The current client's connected player's sprite object
+  private playerSprite: Phaser.GameObjects.Sprite | undefined;
+  // The previous tick's aim radians (used to check if aim has changed, before sending an update)
+  private prevAimRad: number = 0;
+
+  constructor() {
+    super("scene-game");
+  }
+
+  init(data: Record<string, any>) {
```

Comment:
```suggestion
  init({ connection, token }: { connection: HathoraConnection, token: string }) {
```

Reply:
Ah, see I tried doing it that way but couldn't figure out the correct TS syntax. Thank you! :)